packages/lit-node-client-nodejs/src/lib/lit-node-client-nodejs.ts

```diff
@@ -385,14 +391,33 @@ export class LitNodeClientNodeJs
         `Storage key "${storageKey}" is missing. Not a problem. Contiune...`
       );
 
+      // Check if a valid session key exists in cache
+      if (
+        sessionKeyCache &&
+        Date.now() - sessionKeyCache.timestamp < TEMP_CACHE_PERIOD
+      ) {
+        log(`[getSessionKey] Returning session key from cache.`);
+        return sessionKeyCache.value;
+      }
+
       // Generate new one
       const newSessionKey = generateSessionKeyPair();
 
       // (TRY) to set to local storage
       try {
         localStorage.setItem(storageKey, JSON.stringify(newSessionKey));
       } catch (e) {
-        console.warn(`Localstorage not available. Not a problem. Contiune...`);
+        log(
+          `[getSessionKey] Localstorage not available.Not a problem.Contiune...`
+        );
+
+        // Store in cache
+        sessionKeyCache = {
+          value: newSessionKey,
+          timestamp: Date.now(),
+        };
+
```

Comment:
should we just set this regardless of if `localstorage` is available or not since we are checking `sessionKeyCache` in the condition above this try catch? so if we do have to get it from local storage we only do so if the in memory cache is deemed out of date?

Reply:
not sure - i actually think is better to separate them, there's a lot of things we're doing automatically for the users and it's getting hard to trace the side effects/errors when they occur.

Reply:
I see what you mean, so if we are able to access from `localstorage` then we will just use it and ignore the cache?